src/audio/drc/drc_math_generic.c

```diff
@@ -238,7 +238,7 @@ inline int32_t drc_inv_fixed(int32_t x, int32_t precision_x, int32_t precision_y
 	if (precision_inv > precision_y)
 		return Q_SHIFT_RND(inv, precision_inv, precision_y);
 	if (precision_inv < precision_y)
-		return Q_SHIFT_LEFT(inv, precision_inv, precision_y);
+		return sat_int32(Q_SHIFT_LEFT((int64_t)inv, precision_inv, precision_y));
```

Comment:
why do we need typecast int64_t ? 
Do we see a case  [dst - src]  higher than int32_t value  ? 
Do we have a case possible of maximum left shift is 31  bits ?